### PR TITLE
Add CardTable2 and initial real use-case data

### PR DESCRIPTION
**Pull request by magicDGS**

### CODE REVIEW

File: tools/src/test/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2MapperTest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import io.github.ygojson.tools.dataprovider.impl.yugipedia.YugipediaTestData;
+import io.github.ygojson.tools.dataprovider.impl.yugipedia.model.CardTable2;
+
+class CardTable2MapperTest {
+
+	@ParameterizedTest
+	@MethodSource(
+		"io.github.ygojson.tools.dataprovider.impl.yugipedia.YugipediaTestData#getParseWikitextTestData"
+	)
+	void testWikitextToCardTable2(
```
Comment (magicDGS) [inline tools/src/test/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2MapperTest.java:16]: We also need a test for what happen with a null value

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		"\n\\| "
+	);
+
+	/**
+	 * Split pattern fore each key=value pair on the CardTable2 content.
+	 */
+	private static final Pattern CARDTABLE2_KEY_VALUE_SPLIT_PATTERN =
+		Pattern.compile(" += ");
+
+	public CardTable2 wikitextToCardTable2(final String wikitext) {
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java:47]: Change the method to, as we would like to be consistent with all mapper methods names:

```suggestion
	public CardTable2 mapWikitextToCardTable2(final String wikitext) {
```

Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java:47]: Also add a javadoc

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	@Mapping(
+		target = "anti_supports_archetypes",
+		source = "anti-supports_archetypes"
+	)
+	@Mapping(target = "m_s_t", source = "m/s/t")
+	@Mapping(target = "card_type", defaultValue = "Monster")
+	@Mapping(target = "ja_name", qualifiedByName = "japanese")
+	@Mapping(target = "ja_pendulum_effect", qualifiedByName = "japanese")
+	@Mapping(target = "ja_lore", qualifiedByName = "japanese")
+	protected abstract CardTable2 toCardTable2(final Map<String, String> asMap);
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java:78]: Change the method to:

```suggestion
	protected abstract CardTable2 mapToCardTable2(final Map<String, String> asMap);
```

Comment (magicDGS) [reply tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java:78]: Also add javadoc

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+			.collect(Collectors.toMap(val -> val[0], val -> val[1]));
+	}
+
+	@Mapping(target = "anti_supports", source = "anti-supports")
+	@Mapping(
+		target = "anti_supports_archetypes",
+		source = "anti-supports_archetypes"
+	)
+	@Mapping(target = "m_s_t", source = "m/s/t")
+	@Mapping(target = "card_type", defaultValue = "Monster")
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java:74]: Add a comment saying why the `card_type` should be initialized as `Monster` if not present

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.stream.Collectors;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+import io.github.ygojson.tools.dataprovider.impl.yugipedia.model.CardTable2;
+
+@Mapper(uses = MarkupStringMapper.class)
+public abstract class CardTable2Mapper {
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2Mapper.java:15]: Add javadoc

File: tools/src/test/resources/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2MapperTest.testWikitextToCardTable2.continuous_spell.approved.txt [hunk 6]
```diff
@@ -0,0 +1,49 @@
... effect_types=Continuous-like, pendulum_effect_types=null, ocg_status=null, tcg_status=null, tcg_speed_duel_status=null, source_card=null, skill_activation=null, fr_skill_activation=null, de_skill_activation=null, it_skill_activation=null, pt_skill_activation=null, es_skill_activation=null, pendulum_effect=null, fr_pendulum_effect=null, de_pendulum_effect=null, it_pendulum_effect=null, pt_pendulum_effect=null, es_pendulum_effect=null, ja_pendulum_effect=null, trans_pendulum_effect=null, zh_pendulum_effect=null, ko_pendulum_effect=null, lore=Any [[Monster Card|monster]] [[sent]] to the [[GY]] is [[banish]]ed instead., fr_lore=Les monstres envoyés au Cimetière sont bannis à la place., de_lore=Jedes Monster, das auf den Friedhof gelegt wird, wird stattdessen verbannt., it_lore=Qualsiasi mostro mandato al Cimitero viene invece bandito., pt_lore=Qualquer monstro enviado para o Cemitério, em vez disso, é banido., es_lore=Cualquier monstruo mandado al Cementerio es desterrado en su lugar., ja_lore=①：このカードが魔法＆罠ゾーンに存在する限り、お互いの墓地へ送られるモンスターは墓地へは行かず除外される。, trans_lore=null, zh_lore=null, ko_lore=①: 이 카드가 마법 & 함정 존에 존재하는 한, 서로의 묘지로 보내지는 몬스터는 묘지로는 가지 않고 제외된다., use=null, database_id=6707, en_sets=
```
Comment (magicDGS) [inline tools/src/test/resources/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2MapperTest.testWikitextToCardTable2.continuous_spell.approved.txt:1]: This output is polluted with a lot of null values and the multiple lines are only separating specific fields. I would rather test as a json, also like that we can document the mismatch on the java model with json-annotations and we can even have our first tool to output the yugipedia-cardtable2-model instead of directly ours. This tool might not be useful for our project (only for debugging purposes), but it might be useful for the community if they want to have the raw yugipedia data.

File: tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/model/CardTable2.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+	MarkupString pt_sets,
+	MarkupString sp_sets,
+	MarkupString jp_sets,
+	MarkupString ja_sets,
+	MarkupString ae_sets,
+	MarkupString tc_sets,
+	MarkupString sc_sets,
+	MarkupString kr_sets,
+	MarkupString supports,
+	//  anti-supports_archetypes - as it is not a valid name
```
Comment (magicDGS) [inline tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/model/CardTable2.java:107]: Change these comments for a json-annotation, which is useful both as documentation and to output something more readable (and for a custom incoming tool for debugging purposes and to provide to the community).

**Review by magicDGS — COMMENTED** [review]
For the tool, we can create a ticket not to forget the idea, even if the focus will go to ourr model parsing first (but it would be a nice to have one).

File: tools/src/test/resources/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2MapperTest.testWikitextToCardTable2.continuous_spell.approved.txt (hunk 6 above)
Comment (magicDGS) [reply tools/src/test/resources/io/github/ygojson/tools/dataprovider/impl/yugipedia/mapper/CardTable2MapperTest.testWikitextToCardTable2.continuous_spell.approved.txt]: Now we are using json for the tests. Also an issue is created to implement the tool: https://github.com/ygojson/ygojson-tools/issues/37
